lib/consul/migrate/client.rb

```diff
@@ -1,47 +1,63 @@
 require 'json'
 require 'net/http'
+require 'consul/migrate/defaults'
+require 'consul/migrate/error'
 
 module Consul
   module Migrate
     class Client
+      # Make options readable to CLI
       attr_reader :options
 
-      def bind_client; options[:bind_client]; end
-      def port;        options[:port];        end
-      def acl_token;   options[:acl_token];   end
-      def base_url;    "http://#{bind_client}:#{port}"; end
+      def bind_client; @options[:bind_client]; end
+      def port;        @options[:port];        end
+
+      def base_url
+        "http://#{bind_client}:#{port}"
+      end
+
+      def http_params
+        { :token => @options[:acl_token] }
+      end
 
       def initialize(options = {})
-        @options = {
-          :bind_client => 'localhost',
-          :port        => 8500
-        }.merge(options)
+        @options = CLIENT_DEFAULTS.merge(symbolize_keys(options))
       end
 
-      # Get all ACLs
+      # GET /v1/acl/list
       def get_acl_list
-        url = "#{base_url}/v1/acl/list?token=#{acl_token}"
-        response = Net::HTTP.get_response(URI.parse(url))
+        uri = URI("#{base_url}/v1/acl/list")
+        uri.query = URI.encode_www_form(http_params)
+        response = Net::HTTP.get_response(uri)
+
+        if !response.kind_of? Net::HTTPSuccess
```

Comment:
`unless` instead of negative if. You might also want to make it a single-line guard clause if it doesn't make the line too long.